command/agent/host/host.go

```diff
@@ -87,20 +87,22 @@ func environment() map[string]string {
 	return env
 }
 
+// DefaultEnvDenyList is the default set of environment variables that are
+// filtered when passing the environment variables of the host to the task.
+var DefaultEnvDenyList = []string{
+	"CONSUL_TOKEN",
+	"CONSUL_HTTP_TOKEN",
+	"VAULT_TOKEN",
+	"NOMAD_LICENSE",
```

Comment:
It's less of a secret, but should [`NOMAD_LICENSE_PATH`](https://learn.hashicorp.com/tutorials/nomad/hashicorp-enterprise-license?in=nomad/enterprise#read-from-a-file-optional) also be included in the list?

Reply:
I'd like to keep this list constrained to secrets/credentials to both ease maintenance and make the behavior as obvious as possible to users. While I can't think of any reason _to_ expose this env var to users, that's not really how we've treated this configuration parameter traditionally. It's less about making sense and purely about security because security is a lot easier to determine. :sweat_smile:

(This implies that if anybody is securing their license key only by assuming the path is secret, they're doing it wrong.)